workers/unity/Packages/com.improbable.gdk.core/Tests/Editmode/GameObjectRepresentation/SpatialOSComponentTests.cs

```diff
@@ -81,5 +82,47 @@ public void TryGetGameObjectForEntity_returns_false_when_entity_does_not_exist()
             Assert.IsFalse(succeeded);
             Assert.IsNull(linkedGameObject);
         }
+
+        [Test]
+        public void TryGetSpatialOSEntityIdForGameObject_returns_true_when_gameobject_is_linked()
```

Comment:
I think this test name is slightly misleading. 

As I understand it - the GameObject isn't properly linked to the underlying entity at this point? 

Which reveals a more important point - there's no easy way to tell without diving into the ECS layer whether the GameObject is actually linked? 

If the GameObject isn't actually linked - shouldn't the method return false?

Reply:
Yeah it feels slightly brittle right now

Reply:
I added some additional checks for now until we get a chance to refactor it. would that be enough, @jamiebrynes7 ?

Reply:
LGTM